plugins/inputs/windows_event_log/wineventlog/wineventlog.go

```diff
@@ -79,16 +83,26 @@ func NewEventLog(name string, levels []string, logGroupName, logStreamName, rend
 		stateManager:  stateManager,
 		retention:     retention,
 
+		gapsToRead: nil,
+
 		done:          make(chan struct{}),
 		resubscribeCh: make(chan struct{}),
+
+		eventAPI: NewWindowsEventAPI(),
```

Comment:
nit: Would it make sense for this to reference a singleton? It doesn't maintain state right? Could be as simple as
```
var winEventAPI = NewWindowsEventAPI()
```